### DISCUSSION TITLE
Color converter and other ideas

### DISCUSSION THREAD

**Discussion by idr4n** [rsayfutdinov/Pikka#43] [Ideas]
Hi,
I just started using your app, the pro version trial, and I really like it. I have used other ones in the past such as sip and colorsnapper2, but I'm on the hunt for a better one.

Two ideas or request that would make this app quite ideal for me:

- The ability to convert a color to different format right after selecting it with the picker. At the moment I think I have to go to the color converter and somehow paste the color there, but it is not quite straightforward. Colorsnapper, for example, gives the option to select a color format right after clicking with the picker. In your app, it could be something like selecting the color and if I open the converter right after, the color should already be there rather than me typing it or pasting it. Similarly, an easy way to select a color from history and choose another format. I didn't find an easy way to do this so far.
- I was hoping that this app does not steal the focus of other apps when using the picker. Colorsnapper also steal the focus while Sip does not. That is, for example, if I have Alfred or Raycast open and trigger the picker, Alfred window would remain open if using Sip, but not with this app or colorsnapper. In my case, many times I need to pick colors from apps that disappear if another app steal the focus.

Thanks in advance for considering my ideas!

**Comment by rsayfutdinov** [comment]
Hi,

Thanks for your ideas!

1) It is possible to change the color format before picking a color. Take a look at the screenshot below. Is that something that can work for you?
2) I'm aware of the focus stealing problem. Unfortunately, one should use the "private" Apple API to fix this issue. Apps that use private APIs are not allowed in Apple AppStores. The only way to fix this is to distribute the app by myself, avoiding AppStore. I will consider this and try to make the app available to download directly from https://pikka.app

<img width="474" alt="Screenshot 2023-02-22 at 08 19 24" src="https://user-images.githubusercontent.com/12187524/220551286-2c299c5d-2a03-4bd7-a590-d331310b0923.png">

**Reply by idr4n** [reply]
Thanks a lot for your prompt reply and addressing my ideas!

Yes, I was aware that I can change the color format before picking the color, but I would like something I can use without using the mouse. Normally I use one single format but then there are times I need to switch quickly to another one back and forth several times in a row, so going back and forth with changing the color format through the menubar kind of slowdown my workflow significantly. I have the same issue with Sip. Colorsnapper2 has a very good way to do this, however colorsnapper crashes sometimes and it also steal the app focus.

Thanks for explaining the focus issue. I understand that this is tied to Apple policies then. I would appreciate if you consider then offering your app avoiding the AppStore.

Thanks again!

**Reply by rsayfutdinov** [reply]
I was able to fix the issue with focus stealing. You can try it in the last update.
